services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -6069,17 +6067,28 @@ private ExecutionStatusWithDetails getIncrementalPushStatus(
       String clusterName,
       String kafkaTopic,
       String incrementalPushVersion,
-      PushMonitor monitor) {
+      PushMonitor monitor,
+      Store store) {
     HelixCustomizedViewOfflinePushRepository cvRepo =
         getHelixVeniceClusterResources(clusterName).getCustomizedViewRepository();
-    if (!usePushStatusStoreToReadServerIncrementalPushStatus) {
-      return monitor.getIncrementalPushStatusAndDetails(kafkaTopic, incrementalPushVersion, cvRepo);
+
+    if (multiClusterConfigs.getControllerConfig(clusterName).usePushStatusStoreForIncrementalPush()
+        && store.isDaVinciPushStatusStoreEnabled()) {
+      return monitor.getIncrementalPushStatusFromPushStatusStore(
+          kafkaTopic,
+          incrementalPushVersion,
+          cvRepo,
+          getPushStatusStoreReader());
     }
-    return monitor.getIncrementalPushStatusFromPushStatusStore(
-        kafkaTopic,
-        incrementalPushVersion,
-        cvRepo,
-        getPushStatusStoreReader());
+
+    if (multiClusterConfigs.getControllerConfig(clusterName).usePushStatusStoreForIncrementalPush()) {
```

Comment:
nit: we can merge that into above check, which to me looks more straightforward. but it is just a suggestion.

Reply:
Done